gravitee-node-secrets/gravitee-node-secrets-service/src/main/java/io/gravitee/node/secrets/service/conf/GraviteeConfigurationSecretResolverDispatcher.java

```diff
@@ -0,0 +1,182 @@
+    }
+
+    @Override
+    public <T extends SecretManagerConfiguration> T readConfiguration(String pluginId, Class<?> configurationClass) {
+        Map<String, Object> configurationProperties = ConfigHelper.removePrefix(
+            EnvironmentUtils.getAllProperties((ConfigurableEnvironment) environment),
+            String.format("%s.%s", SECRETS_CONFIG_KEY, pluginId)
+        );
+
+        try {
+            @SuppressWarnings("unchecked")
+            Constructor<T> constructor = (Constructor<T>) configurationClass.getDeclaredConstructor(Map.class);
+            return constructor.newInstance(configurationProperties);
+        } catch (NoSuchMethodException | InstantiationException | IllegalAccessException | InvocationTargetException e) {
+            throw new SecretManagerConfigurationException(
+                "Could not create configuration class for secret manager: %s".formatted(pluginId),
+                e
+            );
+        }
+    }
+
+    @Override
+    public Maybe<SecretMap> resolve(SecretMount secretMount) throws SecretProviderNotFoundException, SecretManagerException {
+        if (secrets.containsKey(secretMount.location())) {
+            return Maybe.just(secrets.get(secretMount.location()));
+        }
+        return super.resolve(secretMount).doOnSuccess(secretMap -> secrets.put(secretMount.location(), secretMap));
+    }
+
+    /**
+     * Check if the value given can be handled by a provider.
+     *
+     * @param location the URL of a secret
+     * @return true if there is a provider able to handle this URL
+     */
+    public boolean canHandle(String location) {
+        Objects.requireNonNull(location);
+        return (
+            location.startsWith(SecretProvider.PLUGIN_URL_SCHEME) &&
+            enabledProviders().stream().anyMatch(manager -> canProviderHandle(location, manager))
+        );
+    }
+
+    /**
+     * Check if the value given can be handled by a provider and if the URL can to be use to resolve a single value
+     *
+     * @param location the URL of a secret
+     * @return true if they location can return a single secret
+     */
+    public boolean canResolveSingleValue(String location) {
+        Objects.requireNonNull(location);
+        return (
+            location.startsWith(SecretProvider.PLUGIN_URL_SCHEME) &&
+            enabledProviders()
+                .stream()
+                .anyMatch(manager -> {
+                    try {
+                        SecretMount secretMount = toSecretMount(location);
+                        return canProviderHandle(location, manager) && !secretMount.isKeyEmpty();
+                    } catch (IllegalArgumentException | SecretProviderNotFoundException e) {
+                        // URL might not be suitable for resolving property
+                        return false;
+                    }
+                })
+        );
+    }
```

Comment:
It seems that this method has the same logic as the `canHandle`above plus the isKeyEmpty check. Maybe you could reuse the code to avoid duplicate, wdyt ?